sdl_android/src/main/java/com/smartdevicelink/transport/SdlBroadcastReceiver.java

```diff
@@ -222,9 +222,14 @@ public int compare(ResolveInfo resolveInfo, ResolveInfo t1) {
 									return 0;
 								}
 							});
-							String packageName = apps.get(0).activityInfo.packageName;
-							serviceIntent = new Intent();
-							serviceIntent.setComponent(new ComponentName(packageName, packageName +".SdlRouterService"));
+							do {
```

Comment:
Since we don't use the do/while syntax anywhere else in the project I would prefer if we stuck to either simple while or for loops.